### PR TITLE
Drop codemirror-rails and coffee => Rails 6 + 7 support

### PR DESCRIPTION
**Pull request by apepper**
https://github.com/fixlr/codemirror-rails is officially not supporting Rails 6 or Rails 7. But both rails versions still support the asset pipeline if needed.

This PR converts "coffee" to plain js to remove the coffee dependency.

It also inlines the "vendor/" part of `codemirror-rails` so that it can be directly used. So version `CodeMirror 5.16.0` is now included in the package itself.

### CODE REVIEW

File: scrivito_codemirror_editor.gemspec [hunk 1]
```diff
@@ -2,7 +2,7 @@ Gem::Specification.new do |s|
   s.author = 'Scrivito'
   s.description = 'Make CodeMirror available as a Scrivito editor'
   s.email = 'info@scrivito.com'
-  s.files = Dir['READ*', 'LIC*', 'app/**/*', 'lib/**/*']
+  s.files = Dir['READ*', 'LIC*', 'app/**/*', 'lib/**/*', 'vendor/**/*']
```
Comment (dcsaszar) [inline scrivito_codemirror_editor.gemspec:5]: * [ ] The CodeMirror code looks like the addons rely on the main script to be loaded first. If I read this correctly, we should make the order explicit.

**Review by dcsaszar — CHANGES_REQUESTED** [review]
LGTM but one issue

Also:

* [x] Is the LGPL we currently use compatible with the included MIT? If unsure, we could switch to MIT.

File: scrivito_codemirror_editor.gemspec (hunk 1 above)
Comment (apepper) [reply scrivito_codemirror_editor.gemspec:5]: This is how `codemirror-rails` defines its "files":

https://github.com/fixlr/codemirror-rails/blob/cb5e9eee1bac8b8b6933d330591a5f83100094d8/codemirror-rails.gemspec#L13

So I think we're ok with the current approach.

**Comment by apepper** [comment]
> Is the LGPL we currently use compatible with the included MIT? If unsure, we could switch to MIT.

Since `codemirror` is placed in `vendor` and its javascript files have the following headers:

```js
// CodeMirror, copyright (c) by Marijn Haverbeke and others
// Distributed under an MIT license: http://codemirror.net/LICENSE
```

I think mixing them is good enough. But since `scrivito_codemirror_editor` does not contain much code I simply switch to MIT to be on the safe side.

File: scrivito_codemirror_editor.gemspec (hunk 1 above)
Comment (dcsaszar) [reply scrivito_codemirror_editor.gemspec:5]: I see. Ruby even turned to sorted at some point: https://bugs.ruby-lang.org/issues/8709

**Review by dcsaszar — APPROVED** [review]

**Pull request merged**